### PR TITLE
config.skipAliases: also disable vim, git, xfce and gnome3 aliases

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

add possibility to disable aliases with ```config.skipAliases``` introduced in #39686

Small clean-up, no rebuild at all.

cc @matthewbauer

### CODE REVIEW

**Comment by volth** [comment]
@GrahamcOfBorg eval

**Comment by matthewbauer** [comment]
This looks good! Have you been able to get skipAliases to evaluate though?

File: pkgs/desktops/xfce/default.nix [hunk 1]
```diff
@@ -159,6 +159,7 @@ lib.makeScope pkgs.newScope (self: with self; {
 
   xfce4_power_manager_gtk3 = xfce4-power-manager.override { withGtk3 = true; };
 
+} // lib.optionalAttrs (config.skipAliases or false == false) {
```
Comment (matthewbauer) [inline pkgs/desktops/xfce/default.nix]: This is a little confusing to me... Would ```!(config.skipAliases or false)``` be okay?

Comment (volth) [reply pkgs/desktops/xfce/default.nix]: I'd better invert this option (```skipAliases``` -> ```allowAliases```) while no one uses it.
It has negation in the name and almost every its use is prefixed with ```!```

(```if config.skipAliases or false then {} else``` in "top-level/stage.nix" is also an exact equivalent of ```lib.optionalAttrs !(config.skipAliases or false)```)

Comment (matthewbauer) [reply pkgs/desktops/xfce/default.nix]: Yes that probably makes sense. It’s a little bt awkward because it’s not always obvious what the default value is.

**Comment by matthewbauer** [comment]
Ok I think ```allowAliases``` is an okay name. I might wait a little bit to merging just to make sure no one else has objection (at least I'd hate to have to change the name again if there is some other objections).

Related to this: I am interested in improving the documentation for config options (27975694bb0fb6d3731cef60f2e1cc8009872746). I want to eventually to generate all of these config options into the manual similar to how NixOS options works.

**Comment by volth** [comment]
> I might wait a little bit to merging just to make sure no one else has objection

there is still one ```skipAliases``` in staging (in perl-packages.nix),
so let's wait for ```staging```->```master``` merge

**Comment by volth** [comment]
> there is still one skipAliases in staging (in perl-packages.nix), so let's wait for staging->master merge

@matthewbauer done

**Pull request merged**